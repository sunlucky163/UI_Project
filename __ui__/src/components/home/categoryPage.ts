import { Component } from '@Components/component';
import { MainContent } from '@Components/home/mainContent/mainContent';

export class CategoryPage extends Component {
  private selectors = {
    mainContent: '[data-test-name="contentContainer"]',
  };
  async getMainContent(): Promise<MainContent> {
    await this.page.waitFor(this.selectors);
    const mainContent = await this.page.$(this.selectors.mainContent);
    return new MainContent(this.page, mainContent);
  }
}
