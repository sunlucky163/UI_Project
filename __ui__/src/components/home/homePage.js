"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
var component_1 = require("@Components/component");
var config_1 = require("@Core/config");
var mainMenu_1 = require("@Components/home/mainMenu/mainMenu");
var HomePage = /** @class */ (function (_super) {
    __extends(HomePage, _super);
    function HomePage() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.selectors = {
            mainMenu: '[class="b-header__menu-bar"]',
            isHomeEasy: '//div[contains(@class, "homeEasySteps")]',
        };
        return _this;
    }
    HomePage.prototype.open = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.page.goto(new config_1.Config().get('baseUrl'))];
                    case 1:
                        _a.sent();
                        return [2 /*return*/];
                }
            });
        });
    };
    HomePage.prototype.getMainMenu = function () {
        return __awaiter(this, void 0, void 0, function () {
            var isHomeEasyBlock;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.page.waitFor(this.selectors.mainMenu)];
                    case 1:
                        _a.sent();
                        return [4 /*yield*/, this.page.$(this.selectors.mainMenu)];
                    case 2:
                        isHomeEasyBlock = _a.sent();
                        return [2 /*return*/, new mainMenu_1.MainMenu(this.page, isHomeEasyBlock)];
                }
            });
        });
    };
    return HomePage;
}(component_1.Component));
exports.HomePage = HomePage;
//# sourceMappingURL=homePage.js.map