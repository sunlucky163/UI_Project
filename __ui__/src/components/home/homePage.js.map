{"version":3,"file":"homePage.js","sourceRoot":"","sources":["homePage.ts"],"names":[],"mappings":";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;AAAA,mDAAkD;AAClD,uCAAsC;AACtC,+DAA6D;AAE7D;IAA8B,4BAAS;IAAvC;QAAA,qEAeC;QAdW,eAAS,GAAG;YAChB,QAAQ,EAAE,8BAA8B;YACxC,UAAU,EAAE,0CAA0C;SACzD,CAAC;;IAWN,CAAC;IATS,uBAAI,GAAV;;;;4BACI,qBAAM,IAAI,CAAC,IAAI,CAAC,IAAI,CAAC,IAAI,eAAM,EAAE,CAAC,GAAG,CAAC,SAAS,CAAC,CAAC,EAAA;;wBAAjD,SAAiD,CAAC;;;;;KACrD;IAEK,8BAAW,GAAjB;;;;;4BACI,qBAAM,IAAI,CAAC,IAAI,CAAC,OAAO,CAAC,IAAI,CAAC,SAAS,CAAC,QAAQ,CAAC,EAAA;;wBAAhD,SAAgD,CAAC;wBACzB,qBAAM,IAAI,CAAC,IAAI,CAAC,CAAC,CAAC,IAAI,CAAC,SAAS,CAAC,QAAQ,CAAC,EAAA;;wBAA5D,eAAe,GAAG,SAA0C;wBAClE,sBAAO,IAAI,mBAAQ,CAAC,IAAI,CAAC,IAAI,EAAE,eAAe,CAAC,EAAC;;;;KACnD;IACL,eAAC;AAAD,CAAC,AAfD,CAA8B,qBAAS,GAetC;AAfY,4BAAQ"}