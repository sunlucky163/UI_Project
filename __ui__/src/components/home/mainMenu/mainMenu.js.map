{"version":3,"file":"mainMenu.js","sourceRoot":"","sources":["mainMenu.ts"],"names":[],"mappings":";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;AAAA,mDAAkD;AAElD;IAA8B,4BAAS;IAAvC;QAAA,qEAuBC;QAtBW,eAAS,GAAG;YAChB,UAAU,EAAE,0BAA0B;YACtC,SAAS,EAAE,uDAAuD;SACrE,CAAC;;IAmBN,CAAC;IAjBS,6BAAU,GAAhB;;;;;4BACI,qBAAM,IAAI,CAAC,IAAI,CAAC,OAAO,CAAC,IAAI,CAAC,SAAS,CAAC,UAAU,CAAC,EAAA;;wBAAlD,SAAkD,CAAC;wBACjB,qBAAM,IAAI,CAAC,OAAO,CAAC,CAAC,CAAC,IAAI,CAAC,SAAS,CAAC,UAAU,CAAC,EAAA;;wBAA3E,yBAAyB,GAAG,SAA+C;wBACjF,qBAAM,yBAAyB,CAAC,KAAK,EAAE,EAAA;;wBAAvC,SAAuC,CAAC;;;;;KAC3C;IAEK,+BAAY,GAAlB;;;;;4BACI,qBAAM,IAAI,CAAC,IAAI,CAAC,OAAO,CAAC,IAAI,CAAC,SAAS,CAAC,SAAS,CAAC,EAAA;;wBAAjD,SAAiD,CAAC;wBAChC,qBAAM,IAAI,CAAC,OAAO,CAAC,CAAC,CAAC,IAAI,CAAC,SAAS,CAAC,SAAS,CAAC,EAAA;;wBAA1D,SAAS,GAAG,SAA8C;wBAChE,sBAAO,SAAS,EAAC;;;;KACpB;IAEK,iCAAc,GAApB;;;;;4BACI,qBAAM,IAAI,CAAC,IAAI,CAAC,OAAO,CAAC,IAAI,CAAC,SAAS,CAAC,SAAS,CAAC,EAAA;;wBAAjD,SAAiD,CAAC;wBAChC,qBAAM,IAAI,CAAC,OAAO,CAAC,CAAC,CAAC,IAAI,CAAC,SAAS,CAAC,SAAS,CAAC,EAAA;;wBAA1D,SAAS,GAAG,SAA8C;wBAChE,qBAAM,SAAS,CAAC,KAAK,EAAE,EAAA;;wBAAvB,SAAuB,CAAC;;;;;KAC3B;IACL,eAAC;AAAD,CAAC,AAvBD,CAA8B,qBAAS,GAuBtC;AAvBY,4BAAQ"}