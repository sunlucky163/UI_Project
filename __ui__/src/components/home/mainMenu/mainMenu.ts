import { Component } from '@Components/component';

export class MainMenu extends Component {
    private selectors = {
        vanny_menu: '[data-room="vanny_menu"]',
        firstItem: '[class="b-header-main-menu-content-sub-list__item"] a',
    };

    async clickVanny(): Promise<void> {
        await this.page.waitFor(this.selectors.vanny_menu);
        const addAccessoryButtonElement = await this.element.$(this.selectors.vanny_menu);
        await addAccessoryButtonElement.click();
    }

    async getFirstItem() {
        await this.page.waitFor(this.selectors.firstItem);
        const firstItem = await this.element.$(this.selectors.firstItem);
        return firstItem;
    }

    async clickFirstItem(): Promise<void> {
        await this.page.waitFor(this.selectors.firstItem);
        const firstItem = await this.element.$(this.selectors.firstItem);
        await firstItem.click();
    }
}
