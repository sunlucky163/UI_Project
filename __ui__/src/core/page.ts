import * as puppeteer from 'puppeteer';
import { Element } from '@Core/element';
// eslint-disable-next-line no-duplicate-imports
import { NavigationOptions } from 'puppeteer';

export class Page {
    private readonly page: puppeteer.Page;

    constructor(page: puppeteer.Page) {
        this.page = page;
    }

    async goto(
        url: string,
        options = {
            waitUntil: 'domcontentloaded',
        },
    ): Promise<void> {
        // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
        // @ts-ignore
        await this.page.goto(url, options);
    }

    async waitFor(selector, options = { visible: true }, ...args): Promise<void> {
        if (typeof selector === 'number') {
            await this.page.waitFor(selector);
        } else {
            await this.page.waitFor(selector, options, ...args);
        }
    }

    async $(selector: string): Promise<Element> {
        return new Element(await this.page.$(selector));
    }

    async $$(selector: string): Promise<Element[]> {
        const elements = await this.page.$$(selector);
        return elements.map(element => new Element(element));
    }

    async $x(selector: string): Promise<Element> {
        const [element] = await this.$$x(selector);
        return element;
    }

    async $$x(selector: string): Promise<Element[]> {
        const elements = await this.page.$x(selector);
        return elements.map(element => new Element(element));
    }

    async waitForNavigation(
        options: NavigationOptions = { waitUntil: 'domcontentloaded' },
    ): Promise<void> {
        await this.page.waitForNavigation(options);
    }

    url(): string {
        return this.page.url();
    }
}
