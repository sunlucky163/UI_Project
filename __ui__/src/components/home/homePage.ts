import { Component } from '@Components/component';
import { Config } from '@Core/config';
import { MainMenu} from '@Components/home/mainMenu/mainMenu';

export class HomePage extends Component {
    private selectors = {
        mainMenu: '[class="b-header__menu-bar"]',
        isHomeEasy: '//div[contains(@class, "homeEasySteps")]',
    };

    async open(): Promise<void> {
        await this.page.goto(new Config().get('baseUrl'));
    }

    async getMainMenu(): Promise<MainMenu> {
        await this.page.waitFor(this.selectors.mainMenu);
        const isHomeEasyBlock = await this.page.$(this.selectors.mainMenu);
        return new MainMenu(this.page, isHomeEasyBlock);
    }
}
