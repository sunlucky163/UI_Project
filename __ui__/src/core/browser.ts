import * as puppeteer from 'puppeteer';
import { Config } from '@Core/config';
import { Page } from '@Core/page';
import { launchers } from '@Constants/launchers';
import { forEachSeries } from 'p-iteration';

export class Browser {
    private browser: puppeteer.Browser;
    private readonly launcher: puppeteer.LaunchOptions;

    constructor() {
        const launcher: string = new Config().get('launcher');
        this.launcher = launchers[launcher];
    }

    async launch(): Promise<void> {
        this.browser = await puppeteer.launch(this.launcher);
    }

    async getPage(): Promise<Page> {
        const puppeteerPage = await this.browser.newPage();
        await puppeteerPage.emulate({
            viewport: {
                width: 1280,
                height: 10000,
            },
            userAgent:
                'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.142 Safari/537.36',
        });
        const [uselessTab] = await this.browser.pages();
        await uselessTab.close();

        return new Page(puppeteerPage);
    }

    async close(): Promise<void> {
        await forEachSeries(
            await this.browser.pages(),
            async page => await page.close({ runBeforeUnload: true }),
        );
        await this.browser.close();
    }
}
