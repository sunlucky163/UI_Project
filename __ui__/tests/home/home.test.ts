import { Browser } from '@Core/browser';
import { Page } from '@Core/page';
import { HomePage } from '@Components/home/homePage';
import { MainMenu } from '@Components/home/mainMenu/mainMenu';

describe('SEO on Home Page', () => {
    const browser = new Browser();
    let page: Page;
    let homePage: HomePage;
    let mainMenu: MainMenu;

    beforeAll(async () => {
        await browser.launch();
        page = await browser.getPage();
        homePage = new HomePage(page);
        await homePage.open();
        mainMenu = await homePage.getMainMenu();
    });

    afterAll(async () => {
        await browser.close();
    });

    test('check title element', async () => {
        await mainMenu.clickVanny();
        const firstItem = await mainMenu.getFirstItem();
        expect((await firstItem.attributes())['href']).toMatchSnapshot();
        await mainMenu.clickFirstItem();
    })
});
