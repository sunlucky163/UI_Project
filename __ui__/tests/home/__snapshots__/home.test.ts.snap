// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEO on Home Page check title element 1`] = `"/vanny/akrilovye/"`;
